kpi/paginators.py

```diff
@@ -31,6 +31,19 @@ def get_parent_url(self, obj):
         return reverse_lazy('api-root', request=self.context.get('request'))
 
 
+class FastAssetPagination(Paginated):
+    """
+    Pagination class optimized for faster counting for DISTINCT queries on large tables.
+
+    This class overrides the get_count() method to only look at the primary key field, avoiding expensive DISTINCTs
+    comparing several fields. This may not work for queries with lots of joins, especially with one-to-many or
+    many-to-many type relationships.
+    """
+
+    def get_count(self, queryset):
+        return queryset.only('pk').count()
```

Comment:
I thought about that and maybe we want to optimize the count only when a distinct is present? 
Something like that. 

```python 
    def get_count(self, queryset):
        if queryset.query.distinct:
            return queryset.only('pk').count()
        return super().get_count()    
```

What do you think?

Reply:
For the record, I wanted to test if we get some better performance with `values`, but it is uses with `.count()`.  It does not change anything. 

```python 
In [2]: import timeit

In [3]: timeit.timeit('Asset.objects.filter(owner__username="myusername").distinct().only("pk").count()', globals=globals(), number=10000)
Out[3]: 6.9069382949965075

In [4]: timeit.timeit('Asset.objects.filter(owner__username="myusername").distinct().values("pk").count()', globals=globals(), number=10000)
Out[4]: 7.383306294999784

In [5]: timeit.timeit('Asset.objects.filter(owner__username="myusername").distinct().values("pk").count()', globals=globals(), number=10000)
Out[5]: 6.69678862800356

In [6]: timeit.timeit('Asset.objects.filter(owner__username="myusername").distinct().only("pk").count()', globals=globals(), number=10000)
Out[6]: 7.368681169988122
```

Reply:
> I thought about that and maybe we want to optimize the count only when a distinct is present? Something like that.
> 
> ```python
>     def get_count(self, queryset):
>         if queryset.query.distinct:
>             return queryset.only('pk').count()
>         return super().get_count()    
> ```
> 
> What do you think?

I tested this locally by removing and re-adding the 'distinct' clause on the filter and it seemed to work